functions/src/index.ts

```diff
@@ -32,17 +32,29 @@ exports.generateAverageReadingsCsv = functions.pubsub
 
 // Returns true if the current user is authenticated and is an admin user
 async function isAdmin(context: EventContext): Promise<boolean> {
+  console.log("entered isAdmin")
+  console.log("context object")
+  console.log(context)
   if (context && context.auth) {
+    console.log("entered if branch of isAdmin")
+    console.log("fetching user document")
     const userDocument = await firestore
       .doc(`/users/${context.auth.uid}`)
       .get();
+    console.log("user document object")
```

Comment:
You can use `console.log('User document object: ', userDocument) too kinda group them too